### PR TITLE
Updates in gRPC and JDBC client

### PR DESCRIPTION
**Pull request by sangeetaraghu**
Hi @vietj ,

Updated the gRPC and JDBC client docs based on @pmlopes changes.
Also fixed the broken link.

### CODE REVIEW

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc [hunk 1]
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
...
+      ...
+  }
+----
+
+== gRPC APIs support futures
+
+In {VertX} {v4}, the gRPC APIs support futures.  Using these APIs the gRPC plugin generates promisified applications. This makes it easier to reason about inputs and outputs.XXX
+
+The following example shows you XXX in {VertX} {v3x}.
+
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:98]: @pmlopes Can you please help with text here?

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc [hunk 2]
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
...
+// 4.x
+VertxGreeterGrpc.GreeterVertxImplBase service =
+  new VertxGreeterGrpc.GreeterVertxImplBase() {
+      ...
+  }
+----
+
+== gRPC APIs support futures
+
+In {VertX} {v4}, the gRPC APIs support futures.  Using these APIs the gRPC plugin generates promisified applications. This makes it easier to reason about inputs and outputs.XXX
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:95]: @pmlopes Can you please help me rephrase this line?

This makes it easier to reason about inputs and outputs.XXX

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc [hunk 3]
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
...
+        @Override
+        public void sayHello(HelloRequest request, Promise<HelloReply> future) {
+          future.complete(
+              HelloReply.newBuilder().setMessage(request.getName()).build());
+        }
+  }
+----
+
+The following example shows you XXX in {VertX} {v4}.
+
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:113]: @pmlopes Can you please help with text here?

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc [hunk 4]
```diff
@@ -1,8 +1,240 @@
 [id="changes-in-vertx-jdbc-client_{context}"]
 = Changes in {VertX} JDBC client
 
+From {VertX} {v4}, the JDBC client supports SQL client. The SQL common module has also been merged in JDBC client, that is, `io.vertx:vertx-sql-common` merged in `io.vertx:vertx-jdbc-client` module. You will have to remove the `io.vertx:vertx-sql-common` dependency file because `io.vertx:vertx-jdbc-client` will include it. With the merging of SQL common client, all the database APIs have been consolidated into the JDBC client.
+
+In {VertX} {v4}, the SQL client has been updated to include the following clients:XXX
+
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:7]: @pmlopes Does this read correct?

In {VertX} {v4}, the SQL client has been updated to include the following clients:

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc [hunk 5]
```diff
@@ -1,8 +1,240 @@
...
+* Reactive PostgreSQL client. In earlier releases, it included a reactive PostgreSQL client.
+
+* Reactive MySQL client
+* Reactive DB2 client
+* Existing JDBC client that includes the JDBC client API and the SQL client API.
+
+The reactive implementations use the database network protocols. This makes them resource-efficient.
+
+JDBC calls to database are blocking calls. The JDBC client uses worker threads to make these calls non-blocking. XXX
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:16]: @pmlopes Does this read correct?
JDBC calls to database are blocking calls. The JDBC client uses worker threads to make these calls non-blocking.

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc [hunk 6]
```diff
@@ -1,8 +1,240 @@
...
+  .onFailure(e -> {
+    // failed to get a connection
+  })
+  .onSuccess(conn -> {
+    // very important! don't forget to return the connection
+    conn.close();
+  });
+----
+
+== Support for Typsesafe Config XXX
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:76]: @pmlopes Does this section read correct?

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc [hunk 7]
```diff
@@ -1,8 +1,240 @@
...
+        .setUser("sa")
+        // password
+        .setPassword(""),
+      // configure the pool
+      new PoolOptions()
+        .setMaxSize(16)
+    );
+----
+
+NOTE: To use Typesafe Config, you need the `agroal` connection pool. The pool does not expose many configuration options and makes the configuration easy to use.XXX
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:101]: @pmlopes Does this read correct?

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc [hunk 8]
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
 
 == New gRPC compiler plugin
 
-In earlier releases of {VertX}, to compile a gRPC `.proto` descriptor, you had to work with fork of the official gRPC compiler. This fork is maintained by the Eclipse project. Working with the fork was complex.
+In {VertX} {v4}, the module `protoc-gen-grpc-java` is longer available. This module was a fork of the official gRPC compiler. In earlier releases of {VertX}, you had to work with this fork. This fork is maintained by the Eclipse project. Working with the fork was complex.
 
 In previous releases, to work with gRPC, the following details were added to `pom.xml` file.
 
+[source,java]
```
Comment (pmlopes) [inline asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:12]: it should be `[source,xml]`

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc [hunk 9]
```diff
@@ -5,42 +5,123 @@ The following section describes the changes in {VertX} gRPC.
...
+    <!-- NOTE: the gav coordinates point to the 3.x only compiler fork -->
+    <pluginArtifact>io.vertx:protoc-gen-grpc-java:${vertx.grpc.version}:exe:${os.detected.classifier}</pluginArtifact>
+  </configuration>
+  ...
+</plugin>
 ----
 
 In {VertX} {v4}, a new gRPC compiler plugin is available. This plugin uses the official gRPC compiler instead of the fork. To work with the new gRPC plugin, add the following details to `pom.xml` file.
 
+[source,java]
```
Comment (pmlopes) [inline asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:30]: it should be `[source,xml]`

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 2 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:95]: The use of this API allows developers to clearly identify single asynchronous outputs of the service as a `Future` object. The same way, asynchronous streams are represented by a `WriteStream`. Optionally using this style, adapts the gRPC API to follow the same idioms used across all vert.x modules, making it easier to reason on a large vert.x application.

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 1 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:98]: The following example shows the new `Future` api in use, in comparision to what was being generated in {Vertx} {v3x}.

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 3 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:113]: While in {VertX} {v3x} the greet response would be writen to an argument of the function, in {VertX} {v4} the asynchronous response is the method return value. Which reads simpler and the code flow is easy to follow.

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc (hunk 4 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:7]: yes

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc (hunk 5 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:16]: :+1:

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc (hunk 7 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:101]: perhaps replace need with must:

```
NOTE: To use Typesafe Config, you must include the `agroal` connection pool in your project. The pool does not expose many configuration options and makes the configuration easy to use
```

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc (hunk 6 above)
Comment (pmlopes) [reply asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:76]: :+1:

File: asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc (hunk 7 above)
Comment (sangeetaraghu) [reply asciidoc/modules/con_mg_changes-in-vertx-jdbc-client.adoc:101]: Done

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 9 above)
Comment (sangeetaraghu) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:30]: Done

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 1 above)
Comment (sangeetaraghu) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:98]: As discussed @pmlopes used the following sentence:
The following example shows you use of promises in {VertX} {v3x}.

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 3 above)
Comment (sangeetaraghu) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:113]: As discussed @pmlopes used the following sentence:
The following example shows use of futures in {VertX} {v4}.

File: asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc (hunk 2 above)
Comment (sangeetaraghu) [reply asciidoc/modules/con_mg_changes-in-vertx-grpc.adoc:95]: Thank you so much @pmlopes . As discussed using the following sentence:
In {VertX} {v4}, the gRPC APIs support futures. The gRPC plugin generates promisified APIs. These APIs use the standard {VertX} input and output arguments, which makes it easier to create standard {VertX} applications.

**Comment by sangeetaraghu** [comment]
- Updates in gRPC and JDBC client doc. 

- Added draft file for Opentracing

- Added clustering section:
https://github.com/vert-x3/vertx-4-migration-guide/issues/36

- Moved GraphQL under Common Components section:
https://github.com/vert-x3/vertx-4-migration-guide/issues/34

**Pull request merged**